### PR TITLE
KDS Iconography updates

### LINKED ISSUE #7241: "Quiz" Icon Does not appear on Learn > Classes > Assignment Page
### Observed behavior

There's no icon inside the card
…
image

### Expected behavior

There should be "Quiz" icon in the card

### User-facing consequences

…

### Errors and logs

…

### Steps to reproduce

…

### Context

…

### LINKED ISSUE #7250: In Coach Notifications Feeds (Homepage + full activity feed), some icons are missing from notification
@cpauya in your GIF, I'm also seeing some weird spacing here:

image

_Originally posted by @indirectlylit in https://github.com/learningequality/kolibri/issues/7230#issuecomment-653705534_

This might be because the icon for the thing (quiz, resource, etc.) is missing. See screenshot in in second post.

And this is probably related to #7241 , and general changes to `KIcon` interface

### PR DESCRIPTION
**Pull request by nucleogenesis**
### Summary

- Updated Kolibri to use new icon tokens and new icons, removed unused icons and tokens
- Fixed a few misc design issues that I spotted while testing

**IMPORTANT** 

Merge https://github.com/learningequality/kolibri-design-system/pull/55 before you merge this.

Then we must wait for a KDS release including that PR and the last commit to this PR will be bumping the KDS version.

^^ This is done thanks @indirectlylit ! ^^

### Reviewer guidance

Depends on this KDS PR: https://github.com/learningequality/kolibri-design-system/pull/55

Be sure to check that branch out in your local KDS and `yarn link` it: https://github.com/learningequality/kolibri-design-system/#development-in-parallel-with-other-applications

Please note any missing or incorrect icons and hopefully we'll get this wrapped prior to bug bash! 

### References

Fixes https://github.com/learningequality/kolibri/issues/7241
Fixes https://github.com/learningequality/kolibri/issues/7250

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLessonResourceLearnerListPage.vue [hunk 1]
```diff
@@ -141,6 +141,7 @@
         return this.lessonMap[this.$route.params.lessonId];
       },
       resource() {
+        console.log(this.contentMap[this.$route.params.resourceId]);
         return this.contentMap[this.$route.params.resourceId];
```
Comment (micahscopes) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsLessonResourceLearnerListPage.vue]: Is this `console.log` statement intentional?

**Comment by micahscopes** [comment]
I was able to reproduce [this missing icon issue](https://github.com/learningequality/kolibri/issues/7241) while reviewing the PR.  Here are the two relevant stack traces I found:

```
vue.runtime.esm.js:619 [Vue warn]: Error in render: "TypeError: Cannot read property 'icon' of undefined"
found in
---> <KIcon> at node_modules/kolibri-design-system/lib/KIcon/index.vue
       <KButton> at node_modules/kolibri-design-system/lib/buttons-and-links/KButton.vue
         <KIconButton> at node_modules/kolibri-design-system/lib/buttons-and-links/KIconButton.vue
           <SearchBox> at kolibri/plugins/learn/assets/src/views/SearchBox.vue
             <ActionBarSearchBox> at kolibri/plugins/learn/assets/src/views/ActionBarSearchBox.vue
               <KeenUiToolbar> at kolibri/core/assets/src/views/KeenUiToolbar.vue
                 <AppBar> at kolibri/core/assets/src/views/AppBar.vue
                   <ScrollingHeader> at kolibri/core/assets/src/views/CoreBase/ScrollingHeader.vue
                     <CoreBase> at kolibri/core/assets/src/views/CoreBase/index.vue
                       <Root>
```
and 

```
vue.runtime.esm.js:619 [Vue warn]: Error in render: "TypeError: Cannot read property 'icon' of undefined"
found in
---> <KIcon> at node_modules/kolibri-design-system/lib/KIcon/index.vue
       <UiIcon> at node_modules/kolibri-design-system/lib/keen/UiIcon.vue
         <ContentIcon> at kolibri/core/assets/src/views/ContentIcon.vue
           <CardThumbnail> at kolibri/plugins/learn/assets/src/views/ContentCard/CardThumbnail.vue
             <RouterLink>
               <ContentCard> at kolibri/plugins/learn/assets/src/views/ContentCard/index.vue
                 <AssignedExamsCards> at kolibri/plugins/learn/assets/src/views/classes/AssignedExamsCards.vue
                   <ClassAssignmentsPage> at kolibri/plugins/learn/assets/src/views/classes/ClassAssignmentsPage.vue
                     <CoreBase> at kolibri/core/assets/src/views/CoreBase/index.vue
                       <Root>
```

Additional info:
```
TypeError: Cannot read property 'icon' of undefined
    at VueComponent.svgIconComponent (index.vue:51)
    at Watcher.get (vue.runtime.esm.js:4479)
    at Watcher.evaluate (vue.runtime.esm.js:4584)
    at VueComponent.computedGetter [as svgIconComponent] (vue.runtime.esm.js:4836)
    at Object.get (vue.runtime.esm.js:2072)
    at Proxy.render (index.vue?fa14:5)
    at VueComponent.Vue._render (vue.runtime.esm.js:3548)
    at VueComponent.updateComponent (vue.runtime.esm.js:4066)
    at Watcher.get (vue.runtime.esm.js:4479)
    at new Watcher (vue.runtime.esm.js:4468)
```

**Review by micahscopes — CHANGES_REQUESTED** [review]
First the good news.  The dreaded `<h1>hi</h1>` appears to be gone!  :partying_face: 

To review this PR, I first ran `yarn unlink kolibri-design-system` and `yarn install --force`.  I also stopped and started devserver-hot.  For the most part, everything looked right, apart for a few small things.

One is the alignment on these forward/back buttons... To me, it looked a little off relative to the page number info.  I wasn't sure if that's how it's designed (or if it's relevant to these changes), but figured it'd be worth it to share:
image

Another similar thing I noticed was with the "verified" icon in the facility list:
image

I also was able to reproduce the missing icon issue reported in #7241, and pasted some relevant console output in a separate comment.

**Comment by indirectlylit** [comment]
bumped to beta5 - ref: https://github.com/learningequality/kolibri-design-system/pull/65

**Review by indirectlylit — APPROVED** [review]
looking good!

image

image

**Comment by indirectlylit** [comment]
one final build failure:

```
kolibri/plugins/coach/assets/src/views/common/notifications/__test__/NotificationCard.spec.js
  ● NotificationCard component › shows the correct content icon if the assignment is a Quiz

    expect(received).toEqual(expected) // deep equality

    Expected: "quiz"
    Received: "exam"

      114 |       });
      115 |       const contentIcon = wrapper.find({ name: 'ContentIcon' });
    > 116 |       expect(contentIcon.props().kind).toEqual(iconKind);
          |                                        ^
      117 |     }
      118 |   );
      119 | 

      at kolibri/plugins/coach/assets/src/views/common/notifications/__test__/NotificationCard.spec.js:116:40
```

File: kolibri/core/assets/src/views/ContentIcon.vue [hunk 2]
```diff
@@ -125,6 +125,8 @@
         type: String,
         required: true,
         validator(value) {
+          console.log(value);
+          console.log(typeof value);
```
Comment (micahscopes) [inline kolibri/core/assets/src/views/ContentIcon.vue]: a couple of console.log statements here!

**Review by micahscopes — APPROVED** [review]
I checked for the remaining issues and they were gone :astonished: 

(Leaving only a couple of `console.log` statements in their wake :ocean:)

**Pull request merged**